server/backend/mlx_runner.py

```
@@ -1112,17 +400,31 @@ def run_model_enhanced(
+    except Exception:
+        return {"active_memory_gb": 0.0, "peak_memory_gb": 0.0}
+
+
+def check_memory_available(required_gb: float) -> bool:
+    """Pre-flight check before model loading."""
+    try:
+        current_memory = mx.get_active_memory() / 1024**3
+        # Conservative estimate for total memory if not detectable
+        estimated_total = 16.0  # Assume 16GB for modern Macs
+        available = estimated_total - current_memory - 2.0  # 2GB headroom
+        return available >= required_gb
```

Ruff (0.14.11)

Consider moving this statement to an `else` block

(TRY300)